src/foam/dao/JdbcDAO.java

```diff
@@ -0,0 +1,384 @@
+   * @throws SQLException
+   */
+  public void setStatementValues(IndexedPreparedStatement stmt, FObject obj) throws SQLException {
+    Iterator i = properties_.iterator();
+    while ( i.hasNext() ) {
+      PropertyInfo prop = (PropertyInfo) i.next();
+      prop.setStatementValue(stmt, obj);
+    }
+  }
+
+
+  //TODO
+  @Override
+  public FObject remove_(X x, FObject obj) {
+    return null;
+  }
+
+  //TODO
+  @Override
+  public FObject find_(X x, Object id) {
+    return null;
+  }
+
+  //TODO: refine
+  @Override
+  public Sink select_(X x, Sink sink, long skip, long limit, Comparator order, Predicate predicate) {
+    sink = prepareSink(sink);
+
+    Connection               c         = null;
+    IndexedPreparedStatement stmt      = null;
+    ResultSet                resultSet = null;
+
+    try {
+      c = dataSource_.getConnection();
+
+      StringBuilder builder = threadLocalBuilder_.get()
+        .append("select * from ")
+        .append(tableName_);
+
+      if ( predicate != null ) {
+        builder.append(" where ")
+          .append(predicate.createStatement());
+      }
+
+      if ( order != null ) {
+        builder.append(" order by ")
+          .append(order.createStatement());
+      }
+
+      if ( limit > 0 && limit < this.MAX_SAFE_INTEGER ) {
+        builder.append(" limit ").append(limit);
+      }
+
+      if ( skip > 0 && skip < this.MAX_SAFE_INTEGER ) {
+        builder.append(" offset ").append(skip);
+      }
+
+      stmt = new IndexedPreparedStatement(c.prepareStatement(builder.toString()));
+
+      if ( predicate != null ) {
+        predicate.prepareStatement(stmt);
+      }
+
+      resultSet = stmt.executeQuery(); // ???
+      while ( resultSet.next() ) {
+        sink.put(createFObject(resultSet), null);
+      }
+
+      return sink;
+    } catch (Throwable e) {
+      e.printStackTrace();
+      return null;
+    } finally {
+      closeAllQuietly(resultSet, stmt);
+    }
+  }
+
+  /**
+   * Creates an FObject with the appropriate meta-properties.
+   * @param resultSet
+   * @return
+   * @throws Exception
+   */
+  private FObject createFObject(ResultSet resultSet) throws Exception {
+    if ( getOf() == null ) {
+      throw new Exception("`Of` is not set");
+    }
+
+    FObject obj = (FObject) getOf().getObjClass().newInstance();
+    ResultSetMetaData metaData = resultSet.getMetaData();
+
+    int index = 1;
+    Iterator i = properties_.iterator();
+    while ( i.hasNext() ) {
+      // prevent reading out of bounds of result set
+      if ( index > metaData.getColumnCount() )
+        break;
+      // get the property and set the value
+      PropertyInfo prop = (PropertyInfo) i.next();
+      prop.setFromResultSet(resultSet, index++, obj);
+//      prop.set(obj, resultSet.getObject(index++));
+    }
+
+    return obj;
+  }
+
+  /**
+   * Returns list of properties of a metaclass
+   * @param of ClassInfo
+   */
+  private void getObjectProperties(ClassInfo of){
```

Comment:
Generally, making methods or variables private doesn't provide extra protection and makes testing and debugging more difficult. Especially for a method like this which is just a helper method and isn't granting any private access to internal structures.